From 2dc03697c0b9aea5427c323656f25a74cfdf22ba Mon Sep 17 00:00:00 2001
From: Lyubo <lyubodzhamov@gmail.com>
Date: Sun, 30 Dec 2018 11:40:05 -0500
Subject: [PATCH 44/44] Clean up screenshot if nothing selected

If rect is null, run the finisher so the screenshot logic
is cleaned up and prepared for next screenshot. This fixes
BUGBASH-1831.

[basilgello: combined two if-clauses to avoid double else,
  inspired by @BadDaemon]
Signed-off-by: Vasyl Gello <vasek.gello@gmail.com>

Change-Id: Ib46221df2a2a70bc5a901107ca6aa4268a0568f2
---
 .../systemui/screenshot/GlobalScreenshot.java | 22 ++++++++++---------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/screenshot/GlobalScreenshot.java b/packages/SystemUI/src/com/android/systemui/screenshot/GlobalScreenshot.java
index 9f58af927aa..b278ced516d 100644
--- a/packages/SystemUI/src/com/android/systemui/screenshot/GlobalScreenshot.java
+++ b/packages/SystemUI/src/com/android/systemui/screenshot/GlobalScreenshot.java
@@ -668,16 +668,18 @@ class GlobalScreenshot {
                         view.setVisibility(View.GONE);
                         mWindowManager.removeView(mScreenshotLayout);
                         final Rect rect = view.getSelectionRect();
-                        if (rect != null) {
-                            if (rect.width() != 0 && rect.height() != 0) {
-                                // Need mScreenshotLayout to handle it after the view disappears
-                                mScreenshotLayout.post(new Runnable() {
-                                    public void run() {
-                                        takeScreenshot(finisher, statusBarVisible, navBarVisible,
-                                                rect);
-                                    }
-                                });
-                            }
+                        if (rect != null && rect.width() != 0 && rect.height() != 0) {
+                            // Need mScreenshotLayout to handle it after the view disappears
+                            mScreenshotLayout.post(new Runnable() {
+                                public void run() {
+                                    takeScreenshot(finisher, statusBarVisible, navBarVisible,
+                                            rect);
+                                }
+                            });
+                        } else {
+                            // Clean up if nothing is selected
+                            // to initialize a new screenshot later
+                            finisher.run();
                         }
 
                         view.stopSelection();
-- 
2.17.1

